articles/how-to-use-an-api-in-a-vuejs-project/index.md

```diff
@@ -0,0 +1,440 @@
+
+```JavaScript
+this.posts = response.data
+```
+
+The data we get from the request is saved to the 'posts' array which is created in the data property.
+
+```JavaScript
+catch (error) {
+    console.log(error);
+}
+```
+
+If any error occurs during the code execution, the error will be caught and logged in the console.
+
+After requesting data from the API, you will need to call it on a lifecycle hook. Here we will use the `created()` lifecycle hook, this is because we will be able to retrieve sensitive data and events that are active with the `created` hook.
+
+```JavaScript
+<template></template>
+
+<script>
+export default {
+  data() {
+    return {
+      posts: [],
+    };
+  },
+
+  methods: {
+    async getData() {
+      try {
+        const response = await this.$http.get(
+          "http://jsonplaceholder.typicode.com/posts"
+        );
+        // JSON responses are automatically parsed.
+        this.posts = response.data;
+        console.log(posts);
+      } catch (error) {
+        console.log(error);
+      }
+    },
+  },
+
+  created() {
+    this.getData();
+  },
+};
+</script>
+```
+
+We can now display the data in the template by looping through the posts using [v-for](https://vuejs.org/v2/guide/list.html) directive
+
+```JavaScript
+<template>
+  <div>
+    <div v-for="post in posts" v-bind:key="post.id">
+      <h2>{{ post.title }}</h2>
+      <p>{{ post.body }}</p>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      posts: [],
+    };
+  },
+
+  methods: {
+    async getData() {
+      try {
+        const response = await this.$http.get(
+          "http://jsonplaceholder.typicode.com/posts"
+        );
+        this.posts = response.data;
+      } catch (error) {
+        console.log(error);
+      }
+    },
+  },
+
+  created() {
+    this.getData();
+  },
+};
+</script>
+```
+
+#### Using Fetch API method
+
+[Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) is a powerful and flexible method of flexible APIs. It produces a global fetch() method that provides a simple and rational way to fetch resources asynchronously over the network.
+
+To request with the Fetch API, you just have to make the request directly with the `fetch` object and follow all other steps used in the Axios call above.
+
+```JavaScript
+<template>
+  <div>
+    <ul v-for="post in posts" v-bind:key="post.id">
+      <li>{{ post.title }}</li>
+      <p>{{ post.body }}</p>
+    </ul>
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      posts: [],
+    };
+  },
+
+  methods: {
+    async getData() {
+      try {
+        let response = await fetch("http://jsonplaceholder.typicode.com/posts");
+        this.posts = await response.json();;
+      } catch (error) {
```

Comment:
@edidee You made an error in this section. I've fixed it for you this time. Please test your code snippets before you submit your article next time.